packages/@aws-cdk/cdk/lib/synthesis.ts

```diff
@@ -354,6 +354,7 @@ function renderLegacyStacks(artifacts: { [id: string]: cxapi.Artifact }, store:
         environment: { name: artifact.environment.substr('aws://'.length), account: match[1], region: match[2] },
         template,
         metadata: artifact.metadata || {},
+        autoDeploy: artifact.autoDeploy,
```

Comment:
this should also rendered in the non-legacy artifact or we will lose this feature once we deprecate this

Reply:
Where does the code for that live? That's the only place I could find this logic...

Reply:
https://github.com/awslabs/aws-cdk/blob/master/packages/%40aws-cdk/cdk/lib/stack.ts#L472

Reply:
Answered by Elad offline... https://github.com/awslabs/aws-cdk/blob/master/packages/%40aws-cdk/cdk/lib/stack.ts#L472 .

The PR already includes a change in that method.